projects/github-actions/repo-gardening/src/tasks/gather-support-references/index.js

```diff
@@ -0,0 +1,202 @@
+const debug = require( '../../debug' );
+const getComments = require( '../../get-comments' );
+
+/* global GitHub, WebhookPayloadIssue */
+
+/**
+ * Search for a previous comment from this task in our issue.
+ *
+ * @param {Array} issueComments - Array of all comments on that issue.
+ * @returns {Promise<number>} Promise resolving to a number.
+ */
+async function getListComment( issueComments ) {
+	let commentID = 0;
+
+	debug( `gather-support-references: Looking for a previous comment from this task in our issue.` );
+
+	issueComments.map( comment => {
+		if (
+			comment.user.login === 'github-actions[bot]' &&
+			comment.body.includes( '**Support References**' )
+		) {
+			commentID = comment.id;
+		}
+	} );
+
+	return commentID;
+}
+
+/**
+ * Scan the contents of the issue as well as all its comments, get all support references, and add them to an array of references.
+ *
+ * @param {GitHub} octokit      - Initialized Octokit REST client.
+ * @param {string} owner        - Repository owner.
+ * @param {string} repo         - Repository name.
+ * @param {string} number       - Issue number.
+ * @param {Array} issueComments - Array of all comments on that issue.
+ * @returns {Promise<Array>} Promise resolving to an array.
+ */
+async function getIssueReferences( octokit, owner, repo, number, issueComments ) {
+	const ticketReferences = [];
+	const supportIds = [];
+	const referencesRegexP = /[0-9]*-(?:chat|hc|zen|zd)/gim;
+
+	debug( `gather-support-references: Getting references from comments.` );
+	issueComments.map( comment => {
+		if (
+			comment.user.login !== 'github-actions[bot]' ||
+			! comment.body.includes( '**Support References**' )
+		) {
+			ticketReferences.push( ...comment.body.matchAll( referencesRegexP ) );
+		}
+	} );
+
+	debug( `gather-support-references: Getting references from issue body.` );
+	const {
+		data: { body },
+	} = await octokit.rest.issues.get( {
+		owner: owner.login,
+		repo,
+		issue_number: +number,
+	} );
+	ticketReferences.push( ...body.matchAll( referencesRegexP ) );
+
+	// Buid a first array with only the support IDs we've collected.
+	ticketReferences.map( reference => {
+		supportIds.push( reference[ 0 ] );
+	} );
+
+	// That array can still include duplicates, or references that are not formatted quite properly.
+	// Let's build a final array with unique and correct support IDs.
+	const correctedSupportIds = new Set();
+	supportIds.map( supportId => {
+		// xxx-zen and xxx-hc are the preferred formats for tickets and chats.
+		// xxx-zd and xxx-chat, as well as uppercase versions, are considered as alternate versions.
+		const wrongId = supportId.match( /([0-9]*)-(zd|chat)/i );
+		if ( wrongId ) {
+			const correctedId = `${ wrongId[ 1 ] }-${
+				wrongId[ 2 ].toLowerCase() === 'zd' ? 'zen' : 'hc'
+			}`;
+			correctedSupportIds.add( correctedId );
+		} else {
+			correctedSupportIds.add( supportId.toLowerCase() );
+		}
+	} );
```

Comment:
These two blocks could be merged, process `ticketReferences` into `correctedSupportIds` directly without the intermediate step of `supportIds`.

Reply:
Done in ccc8028e6588f19a217a95326578e7acfce2d855